pg-core/src/java/org/pg/Connection.java

```diff
@@ -204,13 +209,13 @@ private void initTypeMapping() {
 
     }
 
-    private static void setSocketOptions(final Socket socket, final Config config) {
+    private static void setSocketOptions(final AsynchronousSocketChannel channel, final Config config) {
         try {
-            socket.setTcpNoDelay(config.SOTCPnoDelay());
-            socket.setSoTimeout(config.SOTimeout());
-            socket.setKeepAlive(config.SOKeepAlive());
-            socket.setReceiveBufferSize(config.SOReceiveBufSize());
-            socket.setSendBufferSize(config.SOSendBufSize());
+            channel.setOption(StandardSocketOptions.TCP_NODELAY, config.SOTCPnoDelay());
+            //channel.setOption(StandardSocketOptions., config.SOTimeout());
```

Comment:
I tried AsyncSocket object in the past, and it confused me that there is no a timeout option (see below).

Reply:
Yes, although for what its worth, I was not able to get socket timeout to work consistently anyway